tools/revpi-set-mac

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+#
+# SPDX-License-Identifier: GPL-2.0
+#
+# Copyright: 2023 KUNBUS GmbH
+#
+
+iface="$1"
+dev_type="$2"
+mac="$3"
+
+# downcase
+mac="${mac,,}"
+# remove dashes
+mac="${mac//-/}"
+# remove colons
+mac="${mac//:/}"
+
+if [ "$#" != 3 ] ||
+  ! [[ "$mac" =~ ^([0-9a-f]{2})([0-9a-f]{2})([0-9a-f]{2})([0-9a-f]{2})([0-9a-f]{2})([0-9a-f]{2})$ ]] ; then
+    echo 1>&2 "Usage: $(basename "$0") <eth interface> <device type> <mac address>"
+    echo 1>&2 ""
+    echo 1>&2 "Supported devices types are:"
+    echo 1>&2 "  lan9514"
+    exit 1
+fi
+
+case $dev_type in
+    lan9514)
+        magic='0x9500'
+        size=256
+        eeprom_file='/usr/share/revpi/lan9514.bin'
```

Comment:
Instead of a mix of "95xx", "9514" and "9500" numbers, I'd prefer settling on just one of them which is consistently used everywhere.

Reply:
Conceivably, could there be a future need to have RevPi product-specific golden images instead of chip-specific ones?

Reply:
The magic number is set for the device family. The eeprom is imho different for 9512 and 9514, so I think this is OK as is.